.github/actions/setup-tebako/action.yml

```diff
@@ -0,0 +1,171 @@
+# ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+# TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+name: 'Setup tebako'
+description: 'Setup environment for tebako'
+inputs:
+  cache:
+    description: 'Tebako cache mode: off/build/use'
+    required: true
+    type: string
+  cache_path:
+    description: 'Tebako cache root folder'
+    required: true
+    default: '.tebako/deps'
+    type: string
+  cache_ver:
+    description: 'Tebako cache version'
+    required: true
+    default: '0'
+    type: string
+  ruby_version:
+    description: 'Ruby version to install, "env" to use RUBY_VER environment variable'
+    required: true
+    default: 'env'
+    type: string
+  bundler_version:
+    description: 'Bundler version to install'
+    required: true
+    default: 'default'
+    type: string
+  cc:
+    description: C compiler
+    required: false
+    default: 
+    type: string
+  cxx:
+    description: C++ compiler
+    required: false
+    default:
+    type: string
+  verbose:
+    description: Enable verbose output
+    required: false
+    default: off
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup environment
+      if: inputs.cache != 'off'
+      shell: bash
+      run: |
+        cat common.env >> $GITHUB_ENV
+        echo "VERBOSE=${{ inputs.verbose }}" >> $GITHUB_ENV
+
+    - name: Prepare compiler
+      shell: bash
+      run: |
+        OS=$(uname -s)
+        if [ "$OS" = "Linux" ] && [ -e "/etc/lsb-release" ]; then
+            echo "CC=${{ inputs.cc || 'clang-12' }}" >> $GITHUB_ENV
+            echo "CXX=${{ inputs.cxx || 'clang++-12' }}" >> $GITHUB_ENV
+            echo "BUNDLE_BUILD__SASSC=--enable-static-stdlib" >> $GITHUB_ENV
+        elif [ "$OS" = "Linux" ] && [ -e "/etc/alpine-release" ]; then
+            echo "CC=${{ inputs.cc || 'clang' }}" >> $GITHUB_ENV
+            echo "CXX=${{ inputs.cxx || 'clang++' }}" >> $GITHUB_ENV
+            echo "VCPKG_FORCE_SYSTEM_BINARIES=1" >> $GITHUB_ENV
+            echo "ALPINE_VER=$(cat /etc/alpine-release)" >> $GITHUB_ENV
+        elif [ "$OS" = "Darwin" ]; then
+            echo "CC=${{ inputs.cc || 'clang' }}" >> $GITHUB_ENV
+            echo "CXX=${{ inputs.cxx || 'clang++' }}" >> $GITHUB_ENV
+            echo "XCODE_VERSION=13.1" >> $GITHUB_ENV
+        else
+            echo "Unsupported operating system."
+        fi
```

Comment:
I would create "advanced OS" environment variable ("Ubuntu/Alpine/MacOS") on the previous step ("Setup environment") and reuse it.  
Consider the case when we decide to support Fedora as well

Reply:
ADVANCED_OS env has been added